lib/compat/wordpress-6.5/fonts/fonts.php

```diff
@@ -84,6 +84,35 @@ function gutenberg_create_initial_post_types() {
 	);
 }
 
+/**
```

Comment:
The `delete_post` meta capability will need to account for deleting fonts with files attached. This is to account for the file deletion hooks in `wp_delete_post()`.

I'm not in love with the performance aspect of this code when checking the font family post type but can't think of more performant approach. 

```suggestion
/**
 * Modify the `delete_post` meta capability for font post types.
 *
 * For font families and font faces containing attached font files, file
 * system access is required by the user in order to delete posts.
 *
 * @param string[] $caps    The primitive capabilities required for the given capability.
 * @param string   $cap     The capability being checked.
 * @param int      $user_id The user ID.
 * @param array    $args    Context for the capability check.
 * @return string[] The modified primitive capabilities required for the given capability.
 */
function gutenberg_delete_font_post_meta_caps( $caps, $cap, $user_id, $args ) {
	if ( in_array( 'do_not_allow', $caps, true ) ) {
		// It's already known that the user is not allowed to perform the requested capability.
		return $caps;
	}

	if ( 'delete_post' !== $cap ) {
		// This filter is only concerned with the 'delete_post' meta capability.
		return $caps;
	}

	$post = get_post( $args[0] );
	if ( ! $post ) {
		$caps[] = 'do_not_allow';
		return $caps;
	}

	$post_type = get_post_type( $post );

	if ( 'wp_font_face' === $post_type ) {
		// Are there any font files associated with this font face?
		$font_files = get_post_meta( $post->ID, '_wp_font_face_file', false );

		if ( empty( $font_files ) ) {
			/*
			* No font files.
			*
			* The user can delete the post based on the 'delete_post' meta capability.
			*/
			return $caps;
		}

		// The user can only delete the post if they can modify the file system.
		$caps[] = 'upload_fonts';

		return $caps;
	}

	if ( 'wp_font_family' === $post_type ) {
		// Are there any font faces associated with this font family?
		$font_faces = get_children(
			array(
				'post_parent' => $post->ID,
				'post_type'   => 'wp_font_face',
			)
		);

		if ( empty( $font_faces ) ) {
			/*
			 * No font faces.
			 *
			 * The user can delete the post based on the 'delete_post' meta capability.
			 */
			return $caps;
		}

		// If any of the font faces contain files, the user needs to be able to modify the file system.
		foreach ( $font_faces as $font_face ) {
			$font_files = get_post_meta( $font_face->ID, '_wp_font_face_file', false );

			if ( ! empty( $font_files ) ) {
				$caps[] = 'upload_fonts';
				// File system caps are required, so no need to check further.
				break;
			}
		}
		return $caps;
	}

	// Return existing caps if the post type is not a font family or font face.
	return $caps;
}

/**
```